lib/fog/vsphere/requests/compute/vm_take_snapshot.rb

```diff
@@ -5,13 +5,13 @@ class Real
         def vm_take_snapshot(options = {})
           raise ArgumentError, 'instance_uuid is a required parameter' unless options.key? 'instance_uuid'
           raise ArgumentError, 'name is a required parameter' unless options.key? 'name'
+          defaults = {
+            description: '',
+            memory: true,
+            quiesce: false
+          }
           vm = get_vm_ref(options['instance_uuid'])
-          task = vm.CreateSnapshot_Task(
-            name: options['name'],
-            description: options['description'] || '',
-            memory: options['memory'] || true,
-            quiesce: options['quiesce'] || false
-          )
+          task = vm.CreateSnapshot_Task(defaults.merge(options))
```

Comment:
this won't work if I'm correct, options have (or can have at least) string keys.
```suggestion
          task = vm.CreateSnapshot_Task(defaults.merge(options.transform_keys(&:to_sym)))
```
This requires ruby `>= 2.5` but that should not be an issue I believe :thinking:

Reply:
or just have defaults in strings and continue expecting options will have string keys :)

Reply:
The gemspec wants ruby `>=2.0.0`